### PR TITLE
Updated code quality bare except warning code changes in class_singleWorker.py

### PR DESCRIPTION
**Pull request by kdcis**
Added specific exceptions
Added log warning 
Removed except handling after some method changes

### CODE REVIEW

File: src/class_singleWorker.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def stopThread(self):
 
         try:
             queues.workerQueue.put(("stopThread", "data"))
-        except:
+        except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:59]: maybe `queue.Full`

File: src/class_singleWorker.py [hunk 2]
```diff
@@ -148,37 +148,37 @@ def run(self):
...
                 try:
                     self.sendMsg()
-                except:
+                except:  # noqa:E722
                     pass
             elif command == 'sendbroadcast':
                 try:
                     self.sendBroadcast()
-                except:
+                except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:156]: add log

File: src/class_singleWorker.py [hunk 3]
```diff
@@ -148,37 +148,37 @@ def run(self):
...
                     pass
             elif command == 'doPOWForMyV2Pubkey':
                 try:
                     self.doPOWForMyV2Pubkey(data)
-                except:
+                except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:161]: add log

File: src/class_singleWorker.py [hunk 4]
```diff
@@ -148,37 +148,37 @@ def run(self):
...
                 try:
                     self.sendOutOrStoreMyV4Pubkey(data)
-                except:
+                except:  # noqa:E722
                     pass
             elif command == 'sendOnionPeerObj':
                 try:
                     self.sendOnionPeerObj(data)
-                except:
+                except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:176]: add log

File: src/class_singleWorker.py [hunk 5]
```diff
@@ -148,37 +148,37 @@ def run(self):
...
                     pass
             elif command == 'resetPoW':
                 try:
                     proofofwork.resetPoW()
-                except:
+                except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:181]: add log

File: src/class_singleWorker.py [hunk 6]
```diff
@@ -299,7 +299,7 @@ def doPOWForMyV2Pubkey(self, adressHash):
             BMConfigParser().set(
                 myAddress, 'lastpubkeysendtime', str(int(time.time())))
             BMConfigParser().save()
-        except:
+        except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:302]: `configparser.NoSectionError`, can pass

also add a general exception handler and log warning

File: src/class_singleWorker.py [hunk 7]
```diff
@@ -312,7 +312,7 @@ def sendOutOrStoreMyV3Pubkey(self, adressHash):
         """
         try:
             myAddress = shared.myAddressesByHash[adressHash]
-        except:
+        except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:315]: `KeyError`, log info

File: src/class_singleWorker.py [hunk 8]
```diff
@@ -382,7 +382,7 @@ def sendOutOrStoreMyV3Pubkey(self, adressHash):
             BMConfigParser().set(
                 myAddress, 'lastpubkeysendtime', str(int(time.time())))
             BMConfigParser().save()
-        except:
+        except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:385]: add `configparser.NoSectionError`, pass
general error handler, log warn

File: src/class_singleWorker.py [hunk 9]
```diff
@@ -546,7 +546,7 @@ def sendBroadcast(self):
                 # , privEncryptionKeyHex
                 privSigningKeyHex, _, pubSigningKey, pubEncryptionKey = \
                     self._getKeysForAddress(fromaddress)
-            except:
+            except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:549]: replace with `KeyError`

add general error handler that logs warning

File: src/class_singleWorker.py [hunk 10]
```diff
@@ -1120,7 +1120,7 @@ def sendMsg(self):
                 privSigningKeyHex, privEncryptionKeyHex, \
                     pubSigningKey, pubEncryptionKey = self._getKeysForAddress(
                         fromaddress)
-            except:
+            except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:1123]: same, `KeyError` feedback to user, general error log warning

File: src/class_singleWorker.py [hunk 11]
```diff
@@ -1192,7 +1192,7 @@ def sendMsg(self):
                 encrypted = highlevelcrypto.encrypt(
                     payload, "04" + hexlify(pubEncryptionKeyBase256)
                 )
-            except:
+            except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:1195]: add a warn log

File: src/class_singleWorker.py [hunk 12]
```diff
@@ -1243,7 +1243,7 @@ def sendMsg(self):
                     time.time() - powStartTime,
                     sizeof_fmt(nonce / (time.time() - powStartTime))
                 )
-            except:
+            except:  # noqa:E722
```
Comment (PeterSurda) [inline src/class_singleWorker.py:1246]: log warning

File: src/class_singleWorker.py [hunk 13]
```diff
@@ -1327,7 +1327,7 @@ def sendMsg(self):
                     try:
                         apiNotifyPath = BMConfigParser().get(
                             'bitmessagesettings', 'apinotifypath')
-                    except:
```
Comment (PeterSurda) [inline src/class_singleWorker.py:1330]: safeGet, default None

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: src/class_singleWorker.py [hunk 14]
```diff
@@ -148,37 +148,37 @@ def run(self):
             if command == 'sendmessage':
                 try:
                     self.sendMsg()
-                except:
+                except:  # noqa:E722
```
[the remark this answers is not in the document]
Comment (kdcis) [reply src/class_singleWorker.py:151]: Updated

File: src/class_singleWorker.py (hunk 2 above)
Comment (kdcis) [reply src/class_singleWorker.py:156]: Updated

File: src/class_singleWorker.py (hunk 3 above)
Comment (kdcis) [reply src/class_singleWorker.py:161]: Updated

File: src/class_singleWorker.py [hunk 15]
```diff
@@ -148,37 +148,37 @@ def run(self):
...
                 try:
                     self.doPOWForMyV2Pubkey(data)
-                except:
+                except:  # noqa:E722
                     pass
             elif command == 'sendOutOrStoreMyV3Pubkey':
                 try:
                     self.sendOutOrStoreMyV3Pubkey(data)
-                except:
+                except:  # noqa:E722
```
[the remark this answers is not in the document]
Comment (kdcis) [reply src/class_singleWorker.py:166]: Updated

File: src/class_singleWorker.py [hunk 16]
```diff
@@ -148,37 +148,37 @@ def run(self):
...
                     pass
             elif command == 'sendOutOrStoreMyV4Pubkey':
                 try:
                     self.sendOutOrStoreMyV4Pubkey(data)
-                except:
+                except:  # noqa:E722
```
[the remark this answers is not in the document]
Comment (kdcis) [reply src/class_singleWorker.py:171]: Updated

File: src/class_singleWorker.py (hunk 4 above)
Comment (kdcis) [reply src/class_singleWorker.py:176]: Updated

File: src/class_singleWorker.py (hunk 5 above)
Comment (kdcis) [reply src/class_singleWorker.py:181]: Updated

File: src/class_singleWorker.py [hunk 17]
```diff
@@ -243,7 +243,7 @@ def _doPOWDefaults(self, payload, TTL,
                 'PoW took %.1f seconds, speed %s.',
                 delta, sizeof_fmt(nonce / delta)
             )
-        except:  # NameError
+        except:  # noqa:E722  # NameError
```
[the remark this answers is not in the document]
Comment (kdcis) [reply src/class_singleWorker.py:246]: Updated

File: src/class_singleWorker.py (hunk 11 above)
Comment (kdcis) [reply src/class_singleWorker.py:1202]: Updated

File: src/class_singleWorker.py (hunk 12 above)
Comment (kdcis) [reply src/class_singleWorker.py:1254]: Updated

File: src/class_singleWorker.py (hunk 13 above)
Comment (kdcis) [reply src/class_singleWorker.py:1330]: Updated

File: src/class_singleWorker.py (hunk 1 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 6 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 7 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 9 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 10 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (PeterSurda) [reply src/class_singleWorker.py:247]: `"Proof of Work exception"`

File: src/class_singleWorker.py [hunk 18]
```diff
@@ -312,8 +315,9 @@ def sendOutOrStoreMyV3Pubkey(self, adressHash):
         """
         try:
             myAddress = shared.myAddressesByHash[adressHash]
-        except:
+        except KeyError:  # noqa:E722
             # The address has been deleted.
+            self.logger.warning("didn't work")
```
Comment (PeterSurda) [inline src/class_singleWorker.py:320]: `"Can't find %s in myAddressByHash", hexlify(addressHash)"`
I think you need hexlify, try to find out.

File: src/class_singleWorker.py [hunk 19]
```diff
@@ -382,10 +386,12 @@ def sendOutOrStoreMyV3Pubkey(self, adressHash):
             BMConfigParser().set(
                 myAddress, 'lastpubkeysendtime', str(int(time.time())))
             BMConfigParser().save()
-        except:
+        except configparser.NoSectionError:
             # The user deleted the address out of the keys.dat file
             # before this finished.
             pass
+        except:  # noqa:E722
+            self.logger.warning("didn't work")
```
Comment (PeterSurda) [inline src/class_singleWorker.py:394]: more meaningful info

File: src/class_singleWorker.py [hunk 20]
```diff
@@ -546,7 +552,8 @@ def sendBroadcast(self):
                 # , privEncryptionKeyHex
                 privSigningKeyHex, _, pubSigningKey, pubEncryptionKey = \
                     self._getKeysForAddress(fromaddress)
-            except:
+            except KeyError:
+                self.logger.warning("didn't work")
```
Comment (PeterSurda) [inline src/class_singleWorker.py:556]: also

File: src/class_singleWorker.py [hunk 21]
```diff
@@ -1120,7 +1127,8 @@ def sendMsg(self):
                 privSigningKeyHex, privEncryptionKeyHex, \
                     pubSigningKey, pubEncryptionKey = self._getKeysForAddress(
                         fromaddress)
-            except:
+            except KeyError:
+                self.logger.warning("didn't work")
```
Comment (PeterSurda) [inline src/class_singleWorker.py:1131]: too

File: src/class_singleWorker.py [hunk 22]
```diff
@@ -546,7 +552,8 @@ def sendBroadcast(self):
                 # , privEncryptionKeyHex
                 privSigningKeyHex, _, pubSigningKey, pubEncryptionKey = \
                     self._getKeysForAddress(fromaddress)
-            except:
+            except KeyError:
```
Comment (PeterSurda) [inline src/class_singleWorker.py:555]: too narrow

File: src/class_singleWorker.py [hunk 23]
```diff
@@ -1120,7 +1127,8 @@ def sendMsg(self):
                 privSigningKeyHex, privEncryptionKeyHex, \
                     pubSigningKey, pubEncryptionKey = self._getKeysForAddress(
                         fromaddress)
-            except:
+            except KeyError:
```
Comment (PeterSurda) [inline src/class_singleWorker.py:1130]: also too narrow

File: src/class_singleWorker.py [hunk 24]
```diff
@@ -1192,7 +1200,8 @@ def sendMsg(self):
                 encrypted = highlevelcrypto.encrypt(
                     payload, "04" + hexlify(pubEncryptionKeyBase256)
                 )
-            except:
+            except:  # noqa:E722
+                self.logger.warning("didn't work")
```
Comment (PeterSurda) [inline src/class_singleWorker.py:1204]: more meaningful explanation

File: src/class_singleWorker.py [hunk 25]
```diff
@@ -1243,8 +1252,8 @@ def sendMsg(self):
                     time.time() - powStartTime,
                     sizeof_fmt(nonce / (time.time() - powStartTime))
                 )
-            except:
-                pass
+            except:  # noqa:E722
+                self.logger.warning("didn't work")
```
Comment (PeterSurda) [inline src/class_singleWorker.py:1256]: better message

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: src/class_singleWorker.py (hunk 25 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 24 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 21 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 23 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 20 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 22 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 19 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

File: src/class_singleWorker.py (hunk 18 above)
Comment (kdcis) [reply src/class_singleWorker.py]: Updated

Comment (kdcis) [reply src/class_singleWorker.py]: yes, hexlify will better represent addressHash

File: src/class_singleWorker.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (kdcis) [reply src/class_singleWorker.py:247]: Updated

File: src/class_singleWorker.py [hunk 26]
```diff
@@ -1120,7 +1139,14 @@ def sendMsg(self):
...
-            except:
+            except (configparser.NoSectionError, configparser.NoOptionError) as err:
+                self.logger.warning("Section or Option did not found: %s", err)
+            except Exception as err:
+                self.logger.error(
+                    'Error within sendMsg. Could not read'
+                    ' the keys from the keys.dat file for a requested'
+                    ' address. %s\n', err
+                )
                 queues.UISignalQueue.put((
```
Comment (PeterSurda) [inline src/class_singleWorker.py:1150]: this section I would put into the previous except

Comment (PeterSurda) [reply src/class_singleWorker.py:1150]: and in this except, put something like `error, can't send` into the `UISignalQueue`. In the log file, maybe put whole error info.

Comment (PeterSurda) [reply src/class_singleWorker.py:1150]: also in `sendBroadcast`

**Review by PeterSurda — CHANGES_REQUESTED** [review]
please make the requested changes

File: src/class_singleWorker.py (hunk 26 above)
Comment (kdcis) [reply src/class_singleWorker.py:1165]: Updated

**Review by PeterSurda — APPROVED** [review]

**Pull request merged**